dashboard/src/components/api/codenvy-machine-auth.factory.js

```diff
@@ -0,0 +1,66 @@
+/*
+ *  [2015] - [2016] Codenvy, S.A.
+ *  All Rights Reserved.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of Codenvy S.A. and its suppliers,
+ * if any.  The intellectual and technical concepts contained
+ * herein are proprietary to Codenvy S.A.
+ * and its suppliers and may be covered by U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+'use strict';
+
+export class CodenvyMachineAuth {
+
+         /**
+          * Default constructor that is using resource
+          * @ngInject for Dependency injection
+          */
+         constructor($injector, $q) {
+           this.$injector = $window;
+           this.$q = $q;
+           this.tokens = [];
+         }
+
+         requestToken(workspaceId) {
+           return this.$injector.get('$http').get('/api/machine/token/' + workspaceId)
```

Comment:
FYI $http should be retrieved with this.$http and injected in constructor
also we're using $resource instead of $http for dealing with REST API